### PR TITLE
Finish transition to `kotlin.test` of `retrofit` and `fx-coroutines`

### LINKED ISSUE #3180: Refactor NetworkEitherCallAdapterTestSuite from Kotest Plugin to Kotlin-test runtime
Arrow is migrating from the Kotest Multiplatform plugin to Kotlin-test,
not because we don't love Kotest but because Arrow wants to support experimental targets like wasm and be buildable and testable with Kotlin EAP & DEV version such that we can help the Kotlin team prevent regression. We're continue using Kotest property testing, and Kotest assertions.

Arrow needs the help of the community to do this, and the Arrow maintainers will be more than happy to help you through the progress.

This ticket requires following changes in `NetworkEitherCallAdapterTestSuite.kt`. Here is a small list of things to do:

 1. Fork & clone the Arrow repository
 2. Checkout the `arrow-2` branch and create a new branch from it.
 3. Replace all tests with shape `"name of test {"` to camelCase/dromedaryCase `@Test fun nameOfTest() = runTest {` in the test class
 4. Remove `StringSpec` as the super class from the test class
 5. Run `./gradlew spotlessApply`
 6. Commit your changes, and push the new branch to your repo
 7. Create a PR on the Arrow repository from your branch targetting `arrow-2`.

Arrow really appreciates all the help, and looks forward to a bright future together in Kotlin :party:

Thanks in advance!

### LINKED ISSUE #3200: Refactor RaceNTest from Kotest Plugin to Kotlin-test runtime
Arrow is migrating from the Kotest Multiplatform plugin to Kotlin-test,
not because we don't love Kotest but because Arrow wants to support experimental targets like wasm and be buildable and testable with Kotlin EAP & DEV version such that we can help the Kotlin team prevent regression. We're continue using Kotest property testing, and Kotest assertions.

Arrow needs the help of the community to do this, and the Arrow maintainers will be more than happy to help you through the progress.

This ticket requires following changes in `RaceNTest.kt`. Here is a small list of things to do:

 1. Fork & clone the Arrow repository
 2. Checkout the `arrow-2` branch and create a new branch from it.
 3. Replace all tests with shape `"name of test {"` to camelCase/dromedaryCase `@Test fun nameOfTest() = runTest {` in the test class
 4. Remove `StringSpec` as the super class from the test class
 5. Run `./gradlew spotlessApply`
 6. Commit your changes, and push the new branch to your repo
 7. Create a PR on the Arrow repository from your branch targetting `arrow-2`.

Arrow really appreciates all the help, and looks forward to a bright future together in Kotlin :party:

Thanks in advance!

### LINKED ISSUE #3201: Refactor ResourceTest from Kotest Plugin to Kotlin-test runtime
Arrow is migrating from the Kotest Multiplatform plugin to Kotlin-test,
not because we don't love Kotest but because Arrow wants to support experimental targets like wasm and be buildable and testable with Kotlin EAP & DEV version such that we can help the Kotlin team prevent regression. We're continue using Kotest property testing, and Kotest assertions.

Arrow needs the help of the community to do this, and the Arrow maintainers will be more than happy to help you through the progress.

This ticket requires following changes in `ResourceTest.kt`. Here is a small list of things to do:

 1. Fork & clone the Arrow repository
 2. Checkout the `arrow-2` branch and create a new branch from it.
 3. Replace all tests with shape `"name of test {"` to camelCase/dromedaryCase `@Test fun nameOfTest() = runTest {` in the test class
 4. Remove `StringSpec` as the super class from the test class
 5. Run `./gradlew spotlessApply`
 6. Commit your changes, and push the new branch to your repo
 7. Create a PR on the Arrow repository from your branch targetting `arrow-2`.

Arrow really appreciates all the help, and looks forward to a bright future together in Kotlin :party:

Thanks in advance!

### PR DESCRIPTION
**Pull request by serras**
Fixes #3201
Fixes #3200
Fixes #3180

### CODE REVIEW

File: arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt [hunk 1]
```diff
@@ -466,7 +465,6 @@ public fun <A> Resource<A>.asFlow(): Flow<A> =
  * instead the caller is responsible for correctly invoking the `release` handler at the appropriate time.
  * This API is useful for building inter-op APIs between [Resource] and non-suspending code, such as Java libraries.
  */
-@DelicateCoroutinesApi
```
Comment (i-walker) [inline arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt:469]: why was this deleted?

Comment (serras) [reply arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt:469]: It's no longer needed. Or was that added by the authors of the file?

Comment (i-walker) [reply arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt:469]: Yes, AFAIK we kept it as a safe-guard for user awareness over the ramifications, but if we don't need that anymore, we might want to mention it in the PR.
cc @nomisRev

Comment (serras) [reply arrow-libs/fx/arrow-fx-coroutines/src/commonMain/kotlin/arrow/fx/coroutines/Resource.kt]: I just put it back. Thanks for the info!

**Pull request merged**